### ISSUE TITLE
Possible mistakes

### ISSUE THREAD

**Issue by Hide927** [xiaoleimiao/UDTL_Lable_Inconsistent#2]
* train_utils_open_univ.py line 273
I checked the official UAN code and they get the `domain_prob` by **adding or not adding `.detach()`**, while your code seems to use only `self.AdversarialNet_auxiliary()`, which is non-adversarial because no grl layer is added.  So the adv_loss here might be wrong? `self.AdversarialNet()` should be used here right?

* train_utils_open_univ.py line 330
`>` should be changed to `<`?  Because the weight < threshold means that it belongs to the unknown class.

* train_utils_open_univ.py line 368-415
Code indentation may be wrong？

Thanks for your effort! This code helped me a lot : )

**Comment by xiaoleimiao** [comment]
Thank you for indicating these errors. 
For your first question, we actually followed the official code for the UAN method, but our code has mistakes as you mentioned. Both self.AdversarialNet and self.AdversarialNet_auxiliary should be involved in the calculation of adv_loss. We have corrected and updated the code. : )

**Comment by Hide927** [comment]
> 

Thank you for your reply. I wish you all the best!  : )

**Comment by xiaoleimiao** [comment]
Thank you for your useful suggestions. Best wishes for you too~




该邮件从移动设备发送


&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xiaoleimiao/UDTL_Lable_Inconsistent"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年1月15日(星期天) 下午4:11
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [xiaoleimiao/UDTL_Lable_Inconsistent] Possible mistakes (Issue #2)





   
Thank you for your reply. I wish you all the best!  : )
